### PR TITLE
Simplified implementation of `canUpdate`. Fixed `canUpdate` flagging up when the local version is greater than the store version in development contexts.

### PR DESCRIPTION
**Pull request by vxern**

### CODE REVIEW

File: lib/new_version.dart [hunk 1]
```diff
@@ -27,29 +27,28 @@ class VersionStatus {
...
-      }
-      // print('new_version canUpdate local $localPad store $storePad');
-      if (localPad.compareTo(storePad) < 0)
+    final local = localVersion.split('.').map(int.parse).toList();
+    final store = storeVersion.split('.').map(int.parse).toList();
+
+    // Each consecutive field in the version notation is less significant than the previous one,
+    // therefore only one comparison needs to yield `true` for it to be determined that the store
+    // version is greater than the local version.
+    for (var i = 0; i < store.length; i++) {
```
Comment (timtraversy) [inline lib/new_version.dart:38]: This is still a little fragile - sometimes the store and local versions are the same length but not always.

**Comment by timtraversy** [comment]
I should write some tests, because version comparison can be complicated. Lots of edge cases.

File: lib/new_version.dart (hunk 1 above)
Comment (vxern) [reply lib/new_version.dart:38]: This entire implementation is pretty naive still, because the semver 2.0 specification specifies that there can also be a build and early release version `rc`, while this implementation doesn't really take that into account. This pull request is not to serve to implement the whole scope of the specification, but rather improve the previous implementation.
